From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Fri, 1 Jul 2022 17:21:37 +1000
Subject: [PATCH] Only send latency updates for local players

External players' latency isn't synced, thus making it pointless
and wasting resources to send latency updates for those players

diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 2f76de141615594eb4f7b899b18308e51c2e809c..d4a2f2b2388f165b1092de6904b3aa1c8d854ba1 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1059,7 +1059,8 @@ public abstract class PlayerList {
             for (int i = 0; i < this.players.size(); ++i) {
                 final ServerPlayer target = (ServerPlayer) this.players.get(i);
 
-                target.connection.send(new ClientboundPlayerInfoUpdatePacket(EnumSet.of(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_LATENCY), this.players.stream().filter(new Predicate<ServerPlayer>() {
+                target.connection.send(new ClientboundPlayerInfoUpdatePacket(EnumSet.of(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_LATENCY), this.localPlayers.stream().filter(new Predicate<ServerPlayer>() { // Multipaper - use this.localPlayers instead of this.players
+
                     @Override
                     public boolean test(ServerPlayer input) {
                         return target.getBukkitEntity().canSee(input.getBukkitEntity());
